src/django_components/expression.py

```diff
@@ -72,6 +109,24 @@ def is_aggregate_key(key: str) -> bool:
     return ":" in key and not key.startswith(":")
 
 
+def is_spread_operator(value: Any) -> bool:
+    if not isinstance(value, str) or not value:
+        return False
+
+    return value.startswith("...")
+
+
+# A string that starts with `...1=`, `...29=`, etc
+ESCAPED_SPREAD_OPERATOR_RE = re.compile(r"^\.\.\.\d+=")
+
+
+def is_escaped_spread_operator(value: Any) -> bool:
+    if not isinstance(value, str) or not value:
+        return False
+
+    return bool(ESCAPED_SPREAD_OPERATOR_RE.match(value))
```

Comment:
I don't get this. Why would you ever write "...123"?

Reply:
So the "escaped_spread_operator" is meant to be internal, users are not expected to write `...1=`.

We convert the spread syntax `...dict` to the "escaped/internal" format `...1=dict`, so Django still parses it as a kwarg, and it remains in the original position.

And there's the number, since there can be multiple spread operators, so they would then be `...1=`, `...2=`, etc.

This way it's trivial to merge the kwargs after the spread operator is replaced with actual values, because we don't have to guess the position where it originally was.

I've added this comment and renamed the vars to `internal_spread_operator` in https://github.com/EmilStenstrom/django-components/pull/596/commits/9ba0c8404b329dc6ed30fae538ab17de1e5a54c7

Reply:
Very clever! Thanks for explaining :)